pyuvdata/uvdata/tests/test_mwa_corr_fits.py

```diff
@@ -102,8 +104,67 @@ def test_read_mwa_write_uvfits(tmp_path):
     assert mwa_uv == uvfits_uv
 
 
+def test_read_mwax_write_uvfits(tmp_path):
+    """
+    MWAX correlator fits to uvfits loopback test.
+
+    Read in MWAX correlator files, write out as uvfits, read back in and check
+    for object equality.
+    """
+    # spoof testfile to contain 2 times and 2 freqs
+    spoof_file = str(tmp_path / "mwax_spoof_ch137_000.fits")
+    with fits.open(filelist[12]) as mini1:
+        mini1[0].header["NFINECHS"] = 2
+        mini1[1].data = np.repeat(mini1[1].data, 2, axis=1)
+        mini1[2].data = np.repeat(mini1[2].data, 2, axis=1)
+        extra_dat = np.copy(mini1[1].data)
+        extra_samps = np.copy(mini1[2].data)
+        # mini1.append(fits.HDUList([mini1[1], mini1[2]]))
+        mini1.append(fits.ImageHDU(extra_dat))
+        mini1.append(fits.ImageHDU(extra_samps))
+        mini1[3].header["TIME"] = 1636374472
+        mini1[4].header["TIME"] = 1636374472
+        mini1[3].header["MILLITIM"] = 0
+        mini1[4].header["MILLITIM"] = 0
+        mini1.writeto(spoof_file)
+    mwax_uv = UVData()
+    uvfits_uv = UVData()
+    messages = [
+        "telescope_location is not set",
+        "some coarse channel files were not submitted",
+    ]
+    with uvtest.check_warnings(UserWarning, messages):
+        mwax_uv.read_mwa_corr_fits(
+            [spoof_file, filelist[11]],
+            correct_cable_len=True,
+            phase_to_pointing_center=True,
+        )
+
+    testfile = str(tmp_path / "outtest_MWAXcorr.uvfits")
+    mwax_uv.write_uvfits(testfile, spoof_nonessential=True)
+    uvfits_uv.read_uvfits(testfile)
+
+    assert mwax_uv == uvfits_uv
+
+
 @pytest.mark.filterwarnings("ignore:telescope_location is not set. ")
```

Comment:
why is the telescope location not being set in most of these tests? Is that always the case for these files or something weird about the test files?

Reply:
The telescope location is set by `set_telescope_params`, which generates this warning before setting the telescope location

Reply:
hmm. I'm not sure we want this kind of warning under the normal course of reading files. I'll take a look at that method and see if there's a way to modify it.

Reply:
Looking at your code, it looks like the metafits specifies the ENU positions, does it also specify the center of the array? It seems like it should.

Reply:
If this isn't the correct usage for the `set_telescope_params` function, I can set the telescope location another way.

Reply:
Oops, your comment came in while I was writing. Yes, I can get the positions from the file. I think that somewhere in the early process I got the idea that using `set_telescope_params` was a preferred method.

Reply:
It _should_ be the same. but I worry that if there's a slight difference, we want to make sure that we use the value that the antenna positions are reported with respect to. So I'd favor using the one in the file.

Reply:
Ok, sounds good.

Reply:
Oooh actually those values aren't in the metafits file. When I initially wrote the reader I hard-coded the location values in (using the values from cotter). Then I switched to using the `set_telescope_params` function.

Reply:
Ok, then I think we should modify that method to accept a boolean that turns off that warning. I don't think we want to warn people when there's nothing to worry about.